core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -388,7 +388,7 @@ class Partition(val topic: String,
     val leaderHWIncremented = inWriteLock(leaderIsrUpdateLock) {
       leaderReplicaIfLocal match {
         case Some(leaderReplica) =>
-          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs)
+          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs).filter(r => !r.ignoreIsrCheck)
```

Comment:
@mimaison : Thanks for the update. For trunk, do you think you can improve the patch by initializing lastCaughtUpTimeMsUnderlying to current time instead? Then, we should be able to get rid of ignoreIsrCheck().